include/deal.II/hp/fe_collection.h

```diff
@@ -450,10 +450,35 @@ namespace hp
 
   /* --------------- inline functions ------------------- */
 
+  namespace
+  {
+    template<bool... Types> struct BoolStorage;
+
+    template<class U, class... Types>
+    struct is_base_of_all
+    {
+      static constexpr bool value =
+        std::is_same<BoolStorage<std::is_base_of<U,Types>::value..., true>,
+        BoolStorage<true, std::is_base_of<U,Types>::value...>>::value;
```

Comment:
Ah, clever!

We will have this question in other places as well where we want to use variadic templates. Would you mind moving this into `template_constraints.h` somewhere? (Can be in an internal namespace, or wherever you find it convenient.)